rustls/tests/common/mod.rs

```diff
@@ -284,8 +302,8 @@ pub fn make_server_config_with_kx_groups(
 
 pub fn get_client_root_store(kt: KeyType) -> Arc<RootCertStore> {
     let mut roots = kt.get_chain();
-    // drop server cert
-    roots.drain(0..1);
+    // drop server cert and intermediate
+    roots.drain(0..2);
```

Comment:
This change is worth a bit of discussion. Previously we were constructing the client root store by reading the `client.fullchain` file (containing EE cert, intermediate and root) and then discarding just the EE cert. The net effect is that we build a root store here that contains _both_ the intermediate and the root.

We use this root store when constructing a client certificate verifier for two purposes:

1. For populating a list of subjects we send in the [certificate_authorities](https://www.rfc-editor.org/rfc/rfc8446#section-4.2.4) extension of a `CertificateRequest` message (TLS 1.2 is handled similarly).
2. For populating a set of trust anchors we provide to webpki when verifying the client certificate.

For the first purpose it _might_ make sense to include the intermediate subjects. I'm not 100% sure. Reading the RFC text I think it could go either way.

For the second purpose it doesn't make sense to include the intermediate cert. We end up using it as a trust anchor which isn't appropriate: it's an intermediate certificate itself issued by a trust anchor. If we try path-building with the EE cert, the intermediates, and these trust anchors we never look at the intemediates: our built path is EE->TA (where the TA is the first intermediate). This means checking revocation status of the intermediate is never performed: we consider it a TA and those aren't covered by CRLs.

In this branch to make it feasible to test the intermediate revocation checking I've changed this function to only return the root TA, not the intermediate. As a result one other unit test (`client_cert_resolve`) that expected to see the intermediate subject in the `certificate_authorities` ext has to be updated to only expect the root TA subject.

Reply:
> SGTM. I'll do some tidying.

Done.

>  I could take a shot at breaking this up in a follow-up PR. I think it's distinct enough from the work here that it'd be clearer to handle separately.

Filed a separate issue for this so I can resolve this feedback thread: https://github.com/rustls/rustls/issues/1549